### ISSUE TITLE
[Bug] 2.4.0: `ip` in `MP.GetPlayerIdentifiers(id)` is wrong

### ISSUE THREAD

**Issue by lionkor** [BeamMP/BeamMP-Server#59]
**Fill out general information**
BeamMP-Server Version: v2.4.0

**Describe the bug**
The `ip` field in the table returned by `MP.GetPlayerIdentifiers()` is wrong.

**Comment by Starystars67** [comment]
Does the IP come from the server to display it or from auth servers?

**Comment by lionkor** [comment]
From the server, from the active connection

**Comment by Starystars67** [comment]
Ah okay, phew haha.

**Comment by lionkor** [comment]
Might be a big-endian / little-endian issue

**Comment by EvanMulawski** [comment]
This should work:

```cpp
int get_ip_str(const struct sockaddr *sa, char *strBuf, size_t strBufSize) {
    switch(sa->sa_family) {
        case AF_INET:
            inet_ntop(AF_INET, &(((struct sockaddr_in *)sa)->sin_addr), strBuf, strBufSize);
            break;
        case AF_INET6:
            inet_ntop(AF_INET6, &(((struct sockaddr_in6 *)sa)->sin6_addr), strBuf, strBufSize);
            break;
        default:
            return 1;
    }
    return 0;
}

void TNetwork::Authentication(const TConnection& ClientConnection) {
    auto Client = CreateClient(ClientConnection.Socket);
    char AddrBuf[INET6_ADDRSTRLEN];
    get_ip_str(&ClientConnection.SockAddr, AddrBuf, sizeof(AddrBuf));
    beammp_trace("This thread is ip " + std::string(AddrBuf));
    Client->SetIdentifier("ip", AddrBuf);

// ...
```

Full sample for doing this below (with IPv6 support). The fourth parameter to `inet_ntop` should be the size of the string buffer.

`populate_sockaddr` is just for testing.

```cpp
#include <iostream>
#include <cstring>
#ifdef __WIN32__
#include <winsock2.h>
#include <WS2tcpip.h>
#else
#include <sys/socket.h>
#include <arpa/inet.h>
#endif
#include <netinet/in.h>

using namespace std;

int get_ip_str(const struct sockaddr *sa, char *strBuf, size_t strBufSize) {
    switch(sa->sa_family) {
        case AF_INET:
            inet_ntop(AF_INET, &(((struct sockaddr_in *)sa)->sin_addr), strBuf, strBufSize);
            break;
        case AF_INET6:
            inet_ntop(AF_INET6, &(((struct sockaddr_in6 *)sa)->sin6_addr), strBuf, strBufSize);
            break;
        default:
            return 1;
    }
    return 0;
}

// https://stackoverflow.com/a/48329644
void populate_sockaddr(int af, int port, char addr[],
        struct sockaddr_storage *dst, socklen_t *addrlen) {

    if (af == AF_INET) {
        struct sockaddr_in *dst_in4 = (struct sockaddr_in *) dst;

        *addrlen = sizeof(*dst_in4);
        memset(dst_in4, 0, *addrlen);
        dst_in4->sin_family = af;
        dst_in4->sin_port = htons(port);
        inet_pton(af, addr, &dst_in4->sin_addr);

    } else if (af == AF_INET6) {
        struct sockaddr_in6 *dst_in6 = (struct sockaddr_in6 *) dst;

        *addrlen = sizeof(*dst_in6);
        memset(dst_in6, 0, *addrlen);
        dst_in6->sin6_family = af;
        dst_in6->sin6_port = htons(port);
        // unnecessary because of the memset(): dst_in6->sin6_flowinfo = 0;
        inet_pton(af, addr, &dst_in6->sin6_addr);
    }
}

void test_ipv4() {
    struct sockaddr_storage addr;
    socklen_t addrlen;
    populate_sockaddr(AF_INET, 80, "1.2.3.4", &addr, &addrlen);
    
    char s[INET6_ADDRSTRLEN];
    get_ip_str((struct sockaddr *)&addr, s, sizeof(s));
    cout << s << "\r\n";
}

void test_ipv6() {
    struct sockaddr_storage addr;
    socklen_t addrlen;
    populate_sockaddr(AF_INET6, 80, "aaaa:bbbb:cccc:dddd:eeee:ffff:0000:1111", &addr, &addrlen);
    
    char s[INET6_ADDRSTRLEN];
    get_ip_str((struct sockaddr *)&addr, s, sizeof(s));
    cout << s << "\r\n";
}

int main() {
    test_ipv4();
    test_ipv6();
}
```

## License

MIT-0. Use source license where referenced.

**Comment by lionkor** [comment]
I'll review this, thank you @EvanMulawski! Assuming it all checks out, would you give me permission to introduce parts of this into the codebase?

**Comment by EvanMulawski** [comment]
@lionkor Yes, I added a license section above and references where applicable.

**Comment by lionkor** [comment]
fixed in #114
